### PR TITLE
Changed APIs to fetch the code climate statistics

### PR DESCRIPTION
**Pull request by surajput32**
Currently, we are using one code climate API which is not giving the important & correct metrics to us. So this PR uses different APIs to fetch the required metrics.

### CODE REVIEW

File: lib/tasks/weekly_code_climate_statistics.rake [hunk 1]
```diff
@@ -1,46 +1,111 @@
...
-          metric_split = d["attributes"]["name"].split(".")
-          metric = metric_split.first
-          if keys.include? metric
-            point = d["attributes"]["points"].last
-            if ["loc", "ratings"].include? metric
-              stats[metric][metric_split.last] = point["value"] || 0
+
+      if repo_response.success?
+        stat = repo.code_climate_statistics.new
+        date_interval = (Date.current - 2.week).beginning_of_week..Date.current
```
Comment (anujaware) [inline lib/tasks/weekly_code_climate_statistics.rake:19]: @surajput32 We have to use date range of last week from `beginning_of_week` to `end_of_week`. Here there are chances that we get data of repeated dates.

File: lib/tasks/weekly_code_climate_statistics.rake [hunk 2]
```diff
@@ -1,46 +1,111 @@
...
+            snap_response = HTTParty.get(snapshot_url, headers: headers, timeout: 20)
+          rescue Timeout::Error => e
+            puts "Error: Request Timeout for #{repo.project.name}"
+            next
+          end
+
+          if snap_response.success?
+            snap_details = JSON.parse(snap_response.body)
+            snap_timestamp = snap_details['data']['attributes']['created_at']
+            snap_date = Date.parse(snap_timestamp)
```
Comment (anujaware) [inline lib/tasks/weekly_code_climate_statistics.rake:38]: @surajput32 We have to maintain `snapshot_id` & `snapshot_time` in the record.

File: lib/tasks/weekly_code_climate_statistics.rake [hunk 3]
```diff
@@ -1,46 +1,111 @@
...
+            if snap_date.in? date_interval
+              stat.lines_of_code = snap_details['data']['attributes']['lines_of_code']
+              stat.total_issues = snap_details['data']['meta']['issues_count']
+              stat.technical_debt_ratio =
+                snap_details['data']['meta']['measures']['technical_debt_ratio']['value']
+              issues_url = "#{snapshot_url}/issues"
+              complexity_issues_url = "#{issues_url}?filter[categories]=Complexity"
+              duplication_issues_url = "#{issues_url}?filter[categories]=Duplication"
+              begin
+                c_issues_response =
```
Comment (anujaware) [inline lib/tasks/weekly_code_climate_statistics.rake:48]: @surajput32 Use fullform `complexity_issues_response`

File: lib/tasks/weekly_code_climate_statistics.rake [hunk 4]
```diff
@@ -1,46 +1,111 @@
...
+                  HTTParty.get(complexity_issues_url, headers: headers, timeout: 20)
+                d_issues_response =
+                  HTTParty.get(duplication_issues_url, headers: headers, timeout: 20)
+              rescue Timeout::Error => e
+                puts "Error: Request Timeout for #{repo.project.name}"
+                next
+              end
+
+              c_issues_details = JSON.parse(c_issues_response.body)
```
Comment (anujaware) [inline lib/tasks/weekly_code_climate_statistics.rake:57]: @surajput32 `complexity_issue_stat` (if it contains only statistics) else `complexity_issue_details`

File: lib/tasks/weekly_code_climate_statistics.rake [hunk 5]
```diff
@@ -1,46 +1,111 @@
...
+              d_issues_details = JSON.parse(d_issues_response.body)
+              stat.complexity_issues = c_issues_details['meta']['total_count']
```
Comment (anujaware) [inline lib/tasks/weekly_code_climate_statistics.rake:59]: @surajput32 `total_complexities` & `total_duplications` OR `complexity_count` `duplication_count`
Use variable name which implies that it's a count and not list of objects. Please make this change for everywhere

**Review by anujaware — COMMENTED** [review]
@surajput32 Good work Suraj. Please resolve minor comments and let's release it on production.
cc: @swapnilchincholkar

File: lib/tasks/weekly_code_climate_statistics.rake (hunk 1 above)
Comment (surajput32) [reply lib/tasks/weekly_code_climate_statistics.rake:19]: Done

File: lib/tasks/weekly_code_climate_statistics.rake (hunk 3 above)
Comment (surajput32) [reply lib/tasks/weekly_code_climate_statistics.rake:48]: Done

File: lib/tasks/weekly_code_climate_statistics.rake (hunk 4 above)
Comment (surajput32) [reply lib/tasks/weekly_code_climate_statistics.rake:57]: Done. Keeping `complexity_issue_details`.

File: lib/tasks/weekly_code_climate_statistics.rake (hunk 2 above)
Comment (surajput32) [reply lib/tasks/weekly_code_climate_statistics.rake]: Done

File: lib/tasks/weekly_code_climate_statistics.rake (hunk 5 above)
Comment (surajput32) [reply lib/tasks/weekly_code_climate_statistics.rake]: Done

**Review by anujaware — APPROVED** [review]
@surajput32 Is this tested? Can we release it on production?
cc: @swapnilchincholkar

**Comment by surajput32** [comment]
> @surajput32 Is this tested? Can we release it on production?
> cc: @swapnilchincholkar

@anujaware I tested it on local. I think we can release it on production. Please verify the task time interval & scheduler timing with @swapnilchincholkar .

**Comment by anujaware** [comment]
@swapnilchincholkar Can you please merge and release it on production tomorrow? We will start getting data from this week.

**Review by swapnilchincholkar — APPROVED** [review]

**Pull request merged**